staging/src/k8s.io/client-go/tools/cache/controller_test.go

```diff
@@ -448,3 +449,57 @@ func TestPanicPropagated(t *testing.T) {
 		t.Errorf("timeout: the panic failed to propagate from the controller run method!")
 	}
 }
+
+// TestControllerRunWithStopOptions confirms that a controller stops itself
+// when the underlying reflector hits a list error by starting a controller
+// and in a separatre goroutine and waiting for the controller stops itself
+// before cancelling the context. If the context is stopped before the test
+// timeout, that means the controller stopped itself as expected.
+func TestControllerRunWithStopOptions(t *testing.T) {
+	// waitTime is how long to wait for the stopOnError=false case to pass.
+	// It needs to just be long enough for the underlying reflector to make
+	// its initial list call and error out.
+	//
+	// We begin seeing flakes every couple thousand runs at 10ms, so 100ms
+	// was chosen as an order of magnitude higher and does not seem to ever flake.
+	waitTime := 100 * time.Millisecond
+	table := []struct {
+		stopOnError bool
+	}{
+		{true},
+		{false},
+	}
+	for _, item := range table {
+
+		source := fcache.NewFakeControllerSource()
+		source.Add(&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod1"}})
+		source.ListError = fmt.Errorf("Access Denied")
+
+		_, controller := NewInformer(
+			source,
+			&v1.Pod{},
+			time.Millisecond*100,
+			ResourceEventHandlerFuncs{},
+		)
+		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			// confirm the controller stops running when it hits a list error
+			defer cancel()
+			controller.RunWithStopOptions(ctx, StopOptions{
+				StopOnError: func(err error) bool {
+					return item.stopOnError
+				},
+			})
+		}()
+		select {
+		case <-ctx.Done():
+			if !item.stopOnError {
+				t.Errorf("controller should NOT have stopped when stopOnError is false")
+			}
+		case <-time.After(waitTime):
```

Comment:
waits less than a second (or few) very frequently cause flakes in resource-constrained environments like CI

Reply:
Yea, Yu had the same concerns. I was open to bumping it further, but because the stress results seem to indicate things only got hairy around 10ms, it seemed like leaving it at 100ms was fine. 

But since you're the second (of two) reviewer to express this concern, I feel like I should just bump it to a second and call it a day. Let me know if you want it bumped higher to two or three seconds.